### ISSUE TITLE
Geth and Parity nodes produce forks

### ISSUE THREAD

**Issue by blazejkrzak** [lukso-network/go-ethereum-aura#41]
**Scenario of the fork**
3 validators go ethereum
2 validators parity
Network is bootstrapping
Validators are synchronized and accept each others blocks
At block 34 we can find transactions that were mined by parity0
Transactions are within block 34 in parity1, and parity nodes stay synchronized
But transactions are not present in any of geth node, and geth network rejected block 34
So there was split
3 geth validators vs 2 parity validators
TL:DR;
```invalid receipt root hash```

**Comment by blazejkrzak** [comment]
Geth log
```
ERROR[11-18|13:13:38.411|core/blockchain.go:2329]                
########## BAD BLOCK #########
Chain config: {ChainID: 8995 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: 0 EIP155: 0 EIP158: <nil> Byzantium: <nil> Constantinople: <nil> Petersburg: <nil> Istanbul: <nil>, Muir Glacier: <nil>, YOLO v1: <nil>, Engine: aura}

Number: 34
Hash: 0xa70d666997a09e19cf1d4f17a4b0ac1adc12fc7dc8a49ed46ad97f1e3322b22a
	 0: cumulative: 21400 gas: 21400 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x51ff56395d6ed824267d401f8028e006603828b40d441f7bf61bdc92d97caab1 logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: 44f29b8f486522c1f59c7789df8097f2da79114bd7c2149aa1998b27a0598b4d
	 1: cumulative: 42864 gas: 21464 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x1997ccb258fa202c64bbdf293c395b11c70b228536c58c89f3bdb8a6dd9e65ea logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: d5f871b51c4020b7956341e303f17838cf32313e195797744e8c94e42f592f8a
	 2: cumulative: 64328 gas: 21464 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x61a6d679ae34d495bbdb3c8646724fe2ba97d45e7561b43a86942e544a6d88d6 logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: 4d1fa199cd2bb6fc7acfd77f621f2f5c89fcbd8ce64f7eaa716ca20f4670b2d2
	 3: cumulative: 85792 gas: 21464 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x822af89d85d2e9592675ee8564385cd5f94f3170adc2f787c9488829253bfc85 logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: 2fd5bf9f8065dc73af9bef724242c7a04b788552e0541b2c2427abfedc538108
	 4: cumulative: 107256 gas: 21464 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x92273e6d6f26196102a9f036b32c8e1de2487b876b0c79dc723a55b6f7ac55c3 logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: d587261b28dcfb9c487f0bbc82b473b18de4bb9e3242566fda893cc1653ff159
	 5: cumulative: 128720 gas: 21464 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xadfa6110bd07db5fcd1dd80fc3c1e311bfba11414dc7a0b5f75422e465b95262 logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: 128cf6c45cb5a2959fd41724f68f2101531bfadca961a7ffc9eb2e61ee990eec
	 6: cumulative: 150184 gas: 21464 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x212ef41e7226c257bfdfd6423ffb8aed589930054e89a79e989513d1586e6c3b logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: 234dbfc22d4ca1ee3ffcdb029bc5dd474df4bf12ddec8c56a56a1f345d1583cb
	 7: cumulative: 171648 gas: 21464 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xf6747cd0e26156aaef0d31ec1153d250c5d51bcb9ae3745e18af3b70471cf4d1 logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: 5ee9b9db5b7f9f54ff2abbc2d9ac9b969a5bb184e367213a60498c4c129a3bf0
	 8: cumulative: 193112 gas: 21464 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x54793b527313498236a202d830fdcba8f325b06fd27cd4d12ab10b047544dbaa logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: 199786070b4bf231e852e64fd3493c36f3e5bff825f918d557717f334a162ba8
	 9: cumulative: 214576 gas: 21464 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x52d221ad0d4e23f1a8f1a0179f43e84cff3244bc751d6fc985fe96c374d20688 logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state: 48bb95161a1d0b23df8fc732e47d7a985165887a48d34f960295d6ce02831860


Error: invalid receipt root hash (remote: c22f57886803ac2832099cc6640754d2b3f257837640282b6f4a3427676f2098 local: 32d896af3d6c0fc3e56324dbd567344dc29bff154bec380089948f72c620a01e)
##############################
 
DEBUG[11-18|13:13:38.415|eth/downloader/downloader.go:1597]      Downloaded item processing failed        number=34  hash="a70d66…22b22a" err="invalid receipt root hash (remote: c22f57886803ac2832099cc6640754d2b3f257837640282b6f4a3427676f2098 local: 32d896af3d6c0fc3e56324dbd567344dc29bff154bec380089948f72c620a01e)"
DEBUG[11-18|13:13:38.415|eth/downloader/downloader.go:1137]      Skeleton fill terminated                 err="syncing canceled (requested)"
DEBUG[11-18|13:13:38.415|eth/downloader/downloader.go:1019]      Skeleton chain invalid                   peer=0912bb9b8f405c7e err="syncing canceled (requested)"
DEBUG[11-18|13:13:38.415|eth/downloader/downloader.go:1020]      Header download terminated               peer=0912bb9b8f405c7e
DEBUG[11-18|13:13:38.415|eth/downloader/downloader.go:1163]      Block body download terminated           err="syncing canceled (requested)"
DEBUG[11-18|13:13:38.415|eth/downloader/downloader.go:1189]      Transaction receipt download terminated  err="syncing canceled (requested)"
DEBUG[11-18|13:13:38.415|eth/downloader/downloader.go:598]       Reset ancient limit to zero 
DEBUG[11-18|13:13:38.415|eth/downloader/downloader.go:450]       Synchronisation terminated               elapsed=177.711ms
DEBUG[11-18|13:13:38.415|eth/downloader/downloader.go:598]       Reset ancient limit to zero 
WARN [11-18|13:13:38.415|eth/downloader/downloader.go:343]       Synchronisation failed, dropping peer    peer=0912bb9b8f405c7e err="retrieved hash chain is invalid: invalid receipt root hash (remote: c22f57886803ac2832099cc6640754d2b3f257837640282b6f4a3427676f2098 local: 32d896af3d6c0fc3e56324dbd567344dc29bff154bec380089948f72c620a01e)"
```

**Comment by blazejkrzak** [comment]
It looks like `byzantiumBlock` in genesis file set up to 0 allowed us to run network without forks

**Comment by blazejkrzak** [comment]
One thing to discover is why geth nodes does not include transactions from txpool
